astropy/table/table.py

```diff
@@ -222,7 +222,7 @@ class Table:
         Additional keyword args when converting table-like object.
     """
 
-    meta = MetaData()
+    meta = MetaData(copy=False)
```

Comment:
How do I disable this feature if I don't want to be far from the shallow?

Reply:
One option is an astropy configuration item that controls the behavior here.  The issue with doing it like this is always that having such a configuration means that the way a particular code runs is not predictable because the user configuration can change.

Reply:
It seems like directly assigning meta and expecting / requiring that to be a deep copy would be unusual.  The most common idioms related to meta should be:
```
# Create table and provide meta.  This gets a deep copy, so no change.
t = Table(data, meta=meta)

# Updating existing table with new meta, so no change.
t.meta.update(my_meta)

# Assigning completely new meta.  This is the one that changes.  But if I were
# writing this code I would expect that `t.meta` is just a normal attribute and that
# it now shares object reference with `my_meta`.  With this PR that is the case.
# It would almost be surprising that `id(t.meta) != id(my_meta)` and that a deep
# copy is always forced.
t.meta = my_meta
```
Upshot - I would advocate that we accept and **highlight** the API change, in particular in the change log and the What's New.  If the user requires a deepcopy then explicitly do so in the user code.  That makes it more readable!

Reply:
The reasoning seems good, but then shouldn't it be the same for `Column` (see also comment above)? (Or even for `MetaData` itself...)

Reply:
> The reasoning seems good, but then shouldn't it be the same for Column (see also comment above)?

I had the same thought initially, but this ended up breaking a number of tests.  So that was a small red flag to just start off being more conservative.  I *think* that it should be possible to eventually make the same change for `Column`, but I am now also convinced it will have a bigger code footprint.  So in the interest of chipping away at problems in smaller pieces, I'd propose leaving that for later.

Reply:
And maybe I can take another crack at it.  I don't recall the specifics now, but just that when I made that change there were a lot of failing tests.  I started looking into fixing things and got stuck and gave up.

Reply:
@taldcroft - probably easiest to keep things separate, but could you raise a new issue so we don't forget?